ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js

```diff
@@ -1146,7 +1146,20 @@ if (typeof window.DatePickerWidget === 'undefined') {
      */
 
     setValue(value) {
+      function isDateDefaultFormat(str) {
+        // test if format is 'yyyy-MM-dd'
+        const pattern = /^\d{4}-\d{2}-\d{2}$/;
+        return pattern.test(str);
+      }
       let currDate =  new Date(value);
+      /**
+       * We only test for 'yyyy-MM-dd' because timezone related issue only occurs when date is selected from calendar.
+       * And calendar only sets value in yyyy-MM-dd
+       */
+      if (isDateDefaultFormat(value)) {
```

Comment:
Date will always come in this format, why have you adding this check ? We don't support any other format in saveValue.

Reply:
If a user manually enters some other date format from the input widget

Reply:
For saveValue there is no formatting, you don't need this check